### PR TITLE
feat(amplify-frontend-javascript): generate amplifyconfiguration.json for JS projects

### PR DESCRIPTION
**Pull request by bombguy**
#### Description of changes
To support NextJS SSR and ESM projects, generate an additional file for amplify JS projects

#### Description of how you validated changes
Verified that valid configuration is rendered for amplify js projects manually & e2e testing.

#### Checklist
- [X] PR description included
- [X] `yarn test` passes
- [X] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-e2e-tests/src/__tests__/aws-exports/js-frontend-config.test.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    // amplify init, this will generate aws-exports in the project
+    await initJSProjectWithProfile(appRoot, { name: 'reactapp' });
+
+    // check if amplify init generated project files
+    expect(fs.existsSync(`${appRoot}/src/aws-exports.js`)).toBeTruthy();
+    expect(fs.existsSync(`${appRoot}/src/amplifyconfiguration.json`)).toBeTruthy();
+
+    const amplifyconfiguration = fs.readJSONSync(`${appRoot}/src/amplifyconfiguration.json`);
+    const awsExportsModule = await import(`${appRoot}/src/amplifyconfiguration.json`);
```
Comment (sobolk) [inline packages/amplify-e2e-tests/src/__tests__/aws-exports/js-frontend-config.test.ts:39]: This doesn't seem to be correct path.

File: packages/amplify-e2e-tests/src/__tests__/aws-exports/js-frontend-config.test.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  });
+
+  test('should update both both aws-exports.js & amplifyconfiguration.json', async () => {
+    // add an category
+    await addAuthWithDefault(appRoot);
+    await amplifyPushAuth(appRoot);
+
+    // when aws-exports updates, both files should still exist
+    const amplifyconfiguration = fs.readJSONSync(`${appRoot}/src/amplifyconfiguration.json`);
+    const awsExportsModule = await import(`${appRoot}/src/amplifyconfiguration.json`);
```
Comment (sobolk) [inline packages/amplify-e2e-tests/src/__tests__/aws-exports/js-frontend-config.test.ts:52]: same here

File: packages/amplify-frontend-javascript/index.js [hunk 3]
```diff
@@ -37,7 +37,8 @@ async function createFrontendConfigsAtPath(context, amplifyResources, amplifyClo
   const cloudOutputsForFrontend = amplifyCloudResources.outputsForFrontend;
 
   const amplifyConfig = await getAWSExports(context, newOutputsForFrontend, cloudOutputsForFrontend);
-  await generateAwsExportsAtPath(context, path.join(exportPath, constants.exportsFilename), amplifyConfig);
+  await generateAwsExportsAtPath(context, path.join(exportPath, constants.exportsJSFilename), amplifyConfig);
+  await generateAwsExportsAtPath(context, path.join(exportPath, constants.exportsJSONFilename), amplifyConfig);
```
Comment (sobolk) [inline packages/amplify-frontend-javascript/index.js:41]: is this change correct ?
if `generateAwsExportsAtPath` is rendering `aws_exports.js.ejs` template, why would we do that to `json` file ?

File: packages/amplify-e2e-tests/src/__tests__/aws-exports/js-frontend-config.test.ts (hunk 1 above)
Comment (bombguy) [reply packages/amplify-e2e-tests/src/__tests__/aws-exports/js-frontend-config.test.ts:39]: Updated

File: packages/amplify-e2e-tests/src/__tests__/aws-exports/js-frontend-config.test.ts (hunk 2 above)
Comment (bombguy) [reply packages/amplify-e2e-tests/src/__tests__/aws-exports/js-frontend-config.test.ts:52]: Updated

File: packages/amplify-frontend-javascript/index.js (hunk 3 above)
Comment (bombguy) [reply packages/amplify-frontend-javascript/index.js:41]: Updated. VScode decided not to save file.

**Review by sobolk — APPROVED** [review]
LGTM.

Let's merge this when we have 
1. Green tests
2. Confirmation that change works in sample app (via tagged release)

**Comment by HuiSF** [comment]
> LGTM.
> 
> Let's merge this when we have
> 
> 1. Green tests
> 2. Confirmation that change works in sample app (via tagged release)

Tested with the Next.js sample app that's deployed to Vercel and Amplify Hosting, using the newly generated `amplifyconfiguration.json` file works. 👍

**Review by rtpascual — APPROVED** [review]

**Review by sobolk — APPROVED** [review]

**Review by rtpascual — APPROVED** [review]

**Pull request merged**